### DISCUSSION TITLE
Is the decisions section in the docs up to date?

### DISCUSSION THREAD

**Discussion by SelvamArul** [openmeterio/openmeter#678] [Q&A] [answered]
Hi,

Thanks a lot for open-sourcing this wonderful project. I have a question regarding some inconsistencies in the section.
In the [0001-event-streaming-platform.md](https://github.com/openmeterio/openmeter/blob/d8e7d16bb7176f5641f139852dff6c953a74b859/docs/decisions/0001-event-streaming-platform.md) file, you conclude by saying "We have chosen Kafka with ksqlDB."

However, in the [0005-clickhouse.md](https://github.com/openmeterio/openmeter/blob/d8e7d16bb7176f5641f139852dff6c953a74b859/docs/decisions/0005-clickhouse.md) file, several statements contradict this conclusion:

1. Additionally, our initial choice of the streaming processor, ksqlDB, revealed limitations when scaling for small to medium-sized producers. The constraints stemmed from ksqlDB's limited capacity to run persisted queries per instance and its lack of support for clusterization.
2. Cons: Users on pre 1.0.0 releases need to migrate from ksqlDB
3. Pre-aggregation: Utilization of ClickHouse's AggregatingMergeTree and MaterializedView features to pre-aggregate usage data into tumbling windows. 

Moreover, [this](https://github.com/openmeterio/openmeter/blob/d8e7d16bb7176f5641f139852dff6c953a74b859/internal/streaming/clickhouse_connector/query.go#L137) code agrees with the point 3 above.

Is my understanding that [0001-event-streaming-platform.md](https://github.com/openmeterio/openmeter/blob/d8e7d16bb7176f5641f139852dff6c953a74b859/docs/decisions/0001-event-streaming-platform.md) is not up to date anymore for the current version correct?

**Comment by tothandras** [comment, marked as answer]
@SelvamArul, the decisions docs are versioned with the prefixes (0001, 0005), and we keep older docs instead of removing them for auditability. In this case, we ran into the limitations of ksqlDB and replaced it with ClickHouse for pre-aggregation.
